lib/view_component/test_helpers.rb

```diff
@@ -52,6 +61,7 @@ def rendered_component
     # @param component [ViewComponent::Base, ViewComponent::Collection] The instance of the component to be rendered.
     # @return [Nokogiri::HTML]
     def render_inline(component, **args, &block)
+      @page = nil
```

Comment:
Can you explain why we need to reset `page` here?

Reply:
`page` needs to be cached for `within` to work. If it's not reset here, then calling `render_inline` multiple times in the same test won't work. There are several existing tests that do this. Previously the result of `page` method was not cached.